src/components/ThemeProvider/ThemeProvider.tsx

```diff
@@ -0,0 +1,54 @@
+import { ReactNode, FC } from 'react';
+import {
+  createTheme,
+  ThemeProvider as MuiThemeProvider,
+} from '@mui/material/styles';
+
+interface Props {
+  children: ReactNode;
+}
+
+const ThemeProvider: FC<Props> = function ThemeProvider({ children }) {
+  const theme = createTheme({
```

Comment:
Move me out of component. If it's inside and `ThemeProvider` is rerendered performance of the app may degrade significantly
After this change you don't need this component at all. Just import provider and theme in entry point